lib/system/web/src/keyboard_engine.rs

```diff
@@ -0,0 +1,122 @@
+use std::collections::{HashMap, BTreeSet};
+
+use web_sys::{KeyboardEvent};
+use wasm_bindgen::JsCast;
+use wasm_bindgen::prelude::*;
+
+use crate::*;
+
+type Bindings = HashMap<BTreeSet<u32>, Vec<Box<dyn FnMut()>>>;
```

Comment:
`Vec<Box<dyn FnMut()>>` - should be refactored to some kind of `CallbackRegistry` which should manage the callbacks there. Anyway, I'm not sure if `Vec` is the best thing here as we want to add and remove callbacks. Maybe something like `HashMap<*const (), Box<dyn FnMut()>>` instead (of course hidden in a nice type)?